n_queen_problem/NQueenProblem.java

```diff
@@ -0,0 +1,53 @@
+import java.util.*;
+
+public class NQueenProblem { 
+    private static void placeQueenOnBoard(int Qi, int[] board) {
+        int n = board.length;
```

Comment:
Rather than declaring `n` in each recursion and using more space isn't is better to directly use `board.length`? @singhpratyush Views?